src/vec.h

```diff
@@ -41,36 +46,54 @@ class Vec;
  * implementation that did not consider things like non-trivial
  * constructor/destructor, please keep T trivial.
  */
-template <typename T>
+template <typename T, bool shared>
 class Vec : public VecView<T> {
  public:
-  Vec() {}
+  Vec() : VecView<T>() {
+    if constexpr (shared) count_ = new std::atomic<int>(1);
```

Comment:
need `if constexpr` to avoid type errors, because `count_` is of type `Empty` when `shared = false`.